### PR TITLE
upgrade versions of automake and check

### PR DESCRIPTION
**Pull request by nicogbg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Upgrade versions of automake and check to latest

###### Change Log  

- automake upgraded to 1.16.5
- check upgraded to 0.15.2

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/check/check.spec [hunk 1]
```diff
@@ -1,11 +1,11 @@
-Summary:        Check-0.12.0
+Summary:        Check is a unit testing framework for C
```
Comment (PawelWMS) [inline SPECS/check/check.spec:1]: Nice.:)

File: SPECS/check/check.spec [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 Name:           check
-Version:        0.12.0
-Release:        5%{?dist}
+Version:        0.15.2
+Release:        1%{?dist}
 License:        LGPLv2+
 URL:            https://libcheck.github.io/check/
-#Source0:       https://github.com/libcheck/check/archive/%{version}.tar.gz
-Source0:        %{name}-%{version}.tar.gz
+#Source0:       https://github.com/libcheck/check/archive/refs/tags/%{version}.tar.gz
```
Comment (PawelWMS) [inline SPECS/check/check.spec:7]: I think we no longer need this commented `Source0` with the URL below.

File: cgmanifest.json [hunk 3]
```diff
@@ -666,8 +666,8 @@
         "type": "other",
         "other": {
           "name": "check",
-          "version": "0.12.0",
-          "downloadUrl": "https://github.com/libcheck/check/archive/0.12.0.tar.gz"
+          "version": "0.15.2",
+          "downloadUrl": "https://github.com/libcheck/check/archive/refs/tags/0.15.2.tar.gz"
```
Comment (PawelWMS) [inline cgmanifest.json:670]: Why does this URL differ from the uncommented one from the .spec file? I'm a bit confused. If the actual (working) link is this one, then let's not provide invalid ones in the .spec file.

File: toolkit/scripts/toolchain/container/toolchain-sha256sums [hunk 4]
```diff
@@ -7,7 +7,7 @@ b68c15821baecae48db6a36be83c627a4def8d23979b511761d9057cd78ea336  aarch64-jdk8u1
 9541ad4bdc995af2c9d7e65bb3839be06dcc3f84f440329bf8cd4067d184db61  aarch64-jdk8u181-b13-nashorn.tar.bz2
 f4df9cc0041aae52c8c496e2fb1b9cd7e8f1213937e14104b620290936518b22  aarch64-jdk8u181-b13.tar.bz2
 64ebcec9f8ac5b2487125a86a7760d2591ac9e1d3dbd59489633f9de62a57684  autoconf-2.69.tar.xz
-5d05bb38a23fd3312b10aea93840feec685bdf4a41146e78882848165d3ae921  automake-1.16.1.tar.xz
+07bd24ad08a64bc17250ce09ec56e921d6343903943e99ccf63bbf0705e34605  automake-1.16.5.tar.gz
```
Comment (PawelWMS) [inline toolkit/scripts/toolchain/container/toolchain-sha256sums:10]: I think we should also update the URL inside `toolkit/scripts/toolchain/container/toolchain-remote-wget-list`.

**Review by PawelWMS — CHANGES_REQUESTED** [review]
Looks good, my two asks to still get done:
- Update the link inside `toolkit/scripts/toolchain/container/toolchain-remote-wget-list`.
- Unify the source URLs between the .spec files and `cgmanifest.json` for `check`.

File: toolkit/scripts/toolchain/container/toolchain-sha256sums (hunk 4 above)
Comment (nicogbg) [reply toolkit/scripts/toolchain/container/toolchain-sha256sums:10]: yep good catch, Andrew told me that too off line 
I've fixed it

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**